controllers/webhook/status_updater.go

```diff
@@ -0,0 +1,133 @@
+package webhook
+
+import (
+	"context"
+	pipelinesv1 "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha5"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func completionStateForRunCompletionStatus(rcs common.RunCompletionStatus) *pipelinesv1.CompletionState {
+	switch rcs {
+	case common.RunCompletionStatuses.Succeeded:
+		return &pipelinesv1.CompletionStates.Succeeded
+	case common.RunCompletionStatuses.Failed:
+		return &pipelinesv1.CompletionStates.Failed
+	default:
+		return nil
+	}
+}
+
+type StatusUpdater struct {
+	K8sClient client.Client
+}
+
+func NewStatusUpdater(ctx context.Context, scheme *runtime.Scheme) (StatusUpdater, error) {
+	logger := log.FromContext(ctx)
+	k8sConfig, err := common.K8sClientConfig()
+	if err != nil {
+		logger.Error(err, "Error reading k8s client config")
+		return StatusUpdater{}, err
+	}
+
+	k8sClient, err := client.New(k8sConfig, client.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		logger.Error(err, "Error creating k8s client")
+		return StatusUpdater{}, err
+	}
+	logger.Info("StatusUpdater created")
+	return StatusUpdater{k8sClient}, nil
+}
+
+func (su StatusUpdater) handle(
+	ctx context.Context,
+	event common.RunCompletionEvent,
+) error {
+	if event.RunName != nil {
+		if err := su.completeRun(ctx, event); err != nil {
+			return err
+		}
+	}
+	if event.RunConfigurationName != nil {
+		if err := su.completeRunConfiguration(ctx, event); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (su *StatusUpdater) completeRun(ctx context.Context, event common.RunCompletionEvent) error {
+	if event.RunName.Namespace == "" {
```

Comment:
This seems a random extra check? Dont do the same check in the run configuration function?

Reply:
This is done in the run configuration function also (though it is grouped with `status == succeeded`) but same concept applies here without a namespace being set its could be completely non-deterministic which resource should be updated. ie. `pipeline-1` in `default` namespace and `pipeline-1` in `other` namespace.

Reply:
Yes, was blind and didnt see matching check in run config. Understand reasoning, lets add some logging to the situation

Reply:
this entire file was copy-pasted, with the only difference being now it's got a `handle` function.